crates/sui-core/src/consensus_adapter.rs

```diff
@@ -620,19 +621,29 @@ impl ConsensusAdapter {
             .consensus_message_processed_notify(transaction_key)
             .boxed();
 
+        pin_mut!(processed_waiter);
```

Comment:
My understanding is we pin this in a mutable way so we can later borrow under a `&mut ` reference , right? Would that practically be different than declaring the `processed_waiter` as `let mut processed_waiter` from the beginning? (I am purely asking out of curiosity here as I haven't used `pin` explicitly a lot)